bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java

```diff
@@ -308,38 +332,177 @@ private HostAddress getHostAddress(long id, CaseDbConnection connection) throws
+		PreparedStatement preparedStatement = connection.getPreparedStatement(insertSQL, Statement.NO_GENERATED_KEYS);
 
-			PreparedStatement preparedStatement = connection.getPreparedStatement(insertSQL, Statement.NO_GENERATED_KEYS);
+		preparedStatement.clearParameters();
+		preparedStatement.setLong(1, dnsNameAddress.getId());
+		preparedStatement.setLong(2, ipAddress.getId());
+		preparedStatement.setLong(3, source.getId());
+		if (time != null) {
+			preparedStatement.setLong(4, time);
+		} else {
+			preparedStatement.setNull(4, java.sql.Types.BIGINT);
+		}
+		connection.executeUpdate(preparedStatement);
+		recentHostNameAndIpMappingCache.put(ipAddress.getId(), new Byte((byte)1));
+		recentHostNameAndIpMappingCache.put(dnsNameAddress.getId(), new Byte((byte)1));
+	}
 
-			preparedStatement.clearParameters();
-			preparedStatement.setLong(1, dnsNameAddress.getId());
-			preparedStatement.setLong(2, ipAddress.getId());
-			preparedStatement.setLong(3, source.getId());
-			if (time != null) {
-				preparedStatement.setLong(4, time);
-			} else {
-				preparedStatement.setNull(4, java.sql.Types.BIGINT);
+	/**
+	 * Returns true if addressObjectId is used as either IP or host name
+	 * <br>
+	 * <b>Note:</b> This api call uses a database connection. Do not invoke within a transaction.
+	 * 
+	 * @param address
+	 * @return 
+	 */
+	public boolean existsHostNameAndIpMapping(long addressObjectId) throws TskCoreException {
+
+		Byte isPresent = recentHostNameAndIpMappingCache.getIfPresent(addressObjectId);
+		
+		if(Objects.nonNull(isPresent)){
+			return true;
+		}
+		
+		String queryString = "SELECT count(*) as mappingCount FROM tsk_host_address_dns_ip_map WHERE ip_address_id = ? OR dns_address_id = ? ";
+		 
+		db.acquireSingleUserCaseReadLock();
+		try (CaseDbConnection connection = this.db.getConnection();
+				PreparedStatement ps = connection.getPreparedStatement(queryString, Statement.NO_GENERATED_KEYS);) {
+			ps.clearParameters();
+			ps.setLong(1, addressObjectId);
+			ps.setLong(2, addressObjectId);
+			try (ResultSet rs = ps.executeQuery()) {
+				if (!rs.next()) {
+					return false;
+				} else {
+					boolean status = rs.getLong("mappingCount") > 0;
+					if(status){
+						recentHostNameAndIpMappingCache.put(addressObjectId, new Byte((byte)1));
+					}
+					return status;
+				}
 			}
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error looking up host address / Ip mapping for address = " + addressObjectId, ex);
+		} finally {
+			db.releaseSingleUserCaseReadLock();
+		}
+	}
+	
 
-			connection.executeUpdate(preparedStatement);
+	
+	/**
+	 * Returns ObjectId of HostAddress if it exists.
+	 * <br>
+	 * <b>Note:</b> This api call uses a database connection. Do not invoke
+	 * within a transaction.
+	 *
+	 * @param type
+	 * @param address
+	 *
+	 * @return
+	 *
+	 * @throws TskCoreException
+	 */
+	public Optional<Long> existsHostAddress(HostAddress.HostAddressType type, String address) throws TskCoreException {
```

Comment:
I think it would be more natural if this method were named hostAddressExists().

Reply:
renamed